server_test.go

```diff
@@ -82,6 +82,75 @@ func TestServer_testMode(t *testing.T) {
 	t.Logf("HELLO")
 }
 
+func TestServer_testMode_AutoMTLS(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	closeCh := make(chan struct{})
+	go Serve(&ServeConfig{
+		HandshakeConfig: testVersionedHandshake,
+		VersionedPlugins: map[int]PluginSet{
+			2: testGRPCPluginMap,
+		},
+		GRPCServer: DefaultGRPCServer,
+		Logger:     hclog.NewNullLogger(),
+		Test: &ServeTestConfig{
+			Context:          ctx,
+			ReattachConfigCh: nil,
+			CloseCh:          closeCh,
+		},
+	})
+
+	// Connect!
+	process := helperProcess("test-mtls")
+	c := NewClient(&ClientConfig{
+		Cmd:             process,
+		HandshakeConfig: testVersionedHandshake,
+		VersionedPlugins: map[int]PluginSet{
+			2: testGRPCPluginMap,
+		},
+		AllowedProtocols: []Protocol{ProtocolGRPC},
+		AutoMTLS:         true,
+	})
+	client, err := c.Client()
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	// Grab the impl
+	raw, err := client.Dispense("test")
+	if err != nil {
+		t.Fatalf("err should be nil, got %s", err)
+	}
+
+	tester, ok := raw.(testInterface)
+	if !ok {
+		t.Fatalf("bad: %#v", raw)
+	}
+
+	n := tester.Double(3)
+	if n != 6 {
+		t.Fatal("invalid response", n)
+	}
+
+	// ensure we can make use of bidirectional communication with AutoMTLS
+	// enabled
+	err = tester.Bidirectional()
+	if err != nil {
+		t.Fatal("invalid response", err)
+	}
+
+	// Pinging should work
+	if err := client.Ping(); err != nil {
+		t.Fatalf("should not err: %s", err)
+	}
+
+	c.Kill()
```

Comment:
The test comment in `TestServer_testMode` indicates that this does nothing; is the Kill call needed here?

Reply:
Technically the test will still pass without it. But if we don't call Kill then we won't actually kill the "plugin process" in the test. i.e. the log won't include:

```
2022-04-28T20:24:31.942-0500 [DEBUG] plugin: plugin process exited: path=/var/folders/g1/9xn1l6mx0x1dry5wqm78fjpw0000gq/T/go-build1072771554/b001/go-plugin.test pid=33632
2022-04-28T20:24:31.961-0500 [DEBUG] plugin: plugin exited
```

Reply:
In `TestServer_testMode` the Client.process is nil so the call to c.Kill just returns. I think this is because it sets `Cmd` to nil in the ClientConfig

Reply:
I see. That makes sense actually. Let's keep it!